### PR TITLE
Development

### PR DESCRIPTION
**Pull request by jacobeicher**
This pull request contains module 1. We will address any changes during sprint 3

### CODE REVIEW

File: .github/workflows/linter.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      with:
+        python-version: 3.9
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint
+        pip install pyGitHub
+    - name: Analysing the code with pylint
+      run: |
+        find . -type f -name "*.py" | xargs pylint --exit-zero
```
Comment (klfgit) [inline .github/workflows/linter.yaml]: make sure files end with a newline.

File: plugins/modules/repository_information.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                "url": repo.url,
+                "default_branch": repo.default_branch,
+                "hooks_url": repo.hooks_url,
+                "clone_url": repo.clone_url
+                }
+        output.append(current_repo_dict)
+    if module.check_mode:
+        return result
+
+    module.exit_json(repos=output) #PUTS RESULT INTO result.repos
```
Comment (klfgit) [inline plugins/modules/repository_information.py]: Remove the comment.  What is being done is obvious from the code.

File: plugins/modules/repository_information.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                "owner": repo.owner.login,
+                "description": repo.description,
+                "private": repo.private,
+                "archived": repo.archived,
+                "language": repo.language,
+                "url": repo.url,
+                "default_branch": repo.default_branch,
+                "hooks_url": repo.hooks_url,
+                "clone_url": repo.clone_url
+                }
```
Comment (klfgit) [inline plugins/modules/repository_information.py]: The only difference between these blocks are a few attributes.  Set the common attributes in one block of code then add the variations after. e.g.

```
current_repo_dict = {...}
if ...  : 
    curent_repo_dict[...] = ...
```

File: plugins/modules/repository_information.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    result = dict(
+        changed=False,
+        fact=''
+    )
+    #token usage retrieved from module's variables from playbook
```
Comment (klfgit) [inline plugins/modules/repository_information.py]: delete

File: plugins/modules/repository_information.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+    if(module.params['enterprise_url'] == ''):
+        g = Github(module.params['token'])
+    else:
+        g = Github(module.params['token'], base_url=module.params['enterprise_url'])
+
+    output = []
+
+    #organization name retrieved from module's variables from playbook
```
Comment (klfgit) [inline plugins/modules/repository_information.py]: delete

File: plugins/modules/repository_information.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+                "description": repo.description,
+                "private": repo.private,
+                "archived": repo.archived,
+                "language": repo.language,
+                "url": repo.url,
+                "default_branch": repo.default_branch,
+                "hooks_url": repo.hooks_url,
+                "clone_url": repo.clone_url,
+                "visibility": repo.raw_data["visibility"],
+                "is_template": repo.raw_data["is_template"]
```
Comment (klfgit) [inline plugins/modules/repository_information.py]: Are these not available when an enterprise url is used?

File: tests/integration/integration_test.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+      script: ./targets/repository_information/{{ item }}.py TOKEN ORGANIZATION_NAME ENTERPRISE_URL
+      loop:
+        - repository_information_enterprise
+      args:
+        executable: Python3
+      register: result
+
+    - assert:
+        that:
+          result is changed
```
Comment (klfgit) [inline tests/integration/integration_test.yaml]: add a newline.

File: tests/integration/targets/repository_information/repository_information_enterprise.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+        self.assertEqual(['testing-repo-private', 'testing-repo-internal', 'testing-repo-public'], output)
+        self.assertEqual(type(output), list)
+
+if __name__ == '__main__':
+    if len(sys.argv) > 2:
+        TestAPIConnection.ENTERPRISE_URL = sys.argv.pop()
+        TestAPIConnection.ORGANIZATION_NAME = sys.argv.pop()
+        TestAPIConnection.API_KEY = sys.argv.pop()
+    unittest.main()
```
Comment (klfgit) [inline tests/integration/targets/repository_information/repository_information_enterprise.py]: add a newline.

File: tests/integration/targets/repository_information/repository_information_regular.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+            output.append(repo.name)
+
+        self.assertEqual(['kegz', 'ansible-collection-github'], output)
+        self.assertEqual(type(output), list)
+
+if __name__ == '__main__':
+    if len(sys.argv) > 1:
+        TestAPIConnection.ORGANIZATION_NAME = sys.argv.pop()
+        TestAPIConnection.API_KEY = sys.argv.pop()
+    unittest.main()
```
Comment (klfgit) [inline tests/integration/targets/repository_information/repository_information_regular.py]: add a newline.

File: tests/unit/modules/test_github_repo_return.py [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+        test = Organization()
+        test.get_organization('Good Organization Name')
+        test.get_repos()
+        assert test.name_list == [{'name': 'repo1', 'url': 'github.com/repo1'}, {'name': 'repo2', 'url': 'github.com/repo2'}, {'name': 'repo3', 'url': 'github.com/repo3'}]
+
+    def test_fail_get_repos_returns_incorrect_output(self):
+        test = Organization()
+        test.get_organization('Bad Organization Name')
+        test.get_repos()
+        assert test.name_list != [{'name': 'repo1', 'url': 'github.com/repo1'}, {'name': 'repo2', 'url': 'github.com/repo2'}, {'name': 'repo3', 'url': 'github.com/repo3'}]
```
Comment (klfgit) [inline tests/unit/modules/test_github_repo_return.py]: add a newline.

File: plugins/modules/repository_information.py [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+DOCUMENTATION = '''
+---
+module: repository_info
+short_description: A module that returns information about GitHub repositories
+description:
+  - "A module that fetches information about repositories that a GitHub user has access to inside an organization."
+options:
+    token:
+        description:
+            - GitHub API token used to retrieve information about repositories a user has access to
```
Comment (klfgit) [inline plugins/modules/repository_information.py]: "about repositories a user has access to" -> "about repositories to which a use has access"

File: plugins/modules/repository_information.py [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+        required: true
+        type: str
+    enterprise_url:
+        description:
+            - If using a token from a GitHub Enterprise account, the user must pass an enterprise URL
+        required: false
+        type: str
+    organization_name:
+        description:
+          - The organization that the information is within the scope of.
```
Comment (klfgit) [inline plugins/modules/repository_information.py]: "The organization that the information is within the scope of." -> The organization in which the query will be run.

File: plugins/modules/repository_information.py [hunk 13]
```diff
@@ -0,0 +1,165 @@
...
+        required: true
+        type: str
+author:
+    - Jacob Eicher (@jacobeicher)
+    - Bradley Golski (@bgolski)
+    - Tyler Zwolenik (@TylerZwolenik)
```
Comment (klfgit) [inline plugins/modules/repository_information.py:44]: List all 4 of the team members.

File: plugins/modules/repository_information.py [hunk 14]
```diff
@@ -0,0 +1,165 @@
...
+
+EXAMPLES = '''
+# Pass in an organization name and GitHub API token
+- name: returns information about 
+  repository_info:
+    organization: "senior-design-21-22"
+    github_token: "12345"
+
+
+# Pass in an organization name, GitHub API token
```
Comment (klfgit) [inline plugins/modules/repository_information.py]: ", and enterprise URL' seems to be missing.

File: plugins/modules/repository_information.py [hunk 15]
```diff
@@ -0,0 +1,165 @@
...
+    github_token: "12345"
+    enterprise_url: "<ENTERPRISE_URL>"
+'''
+
+RETURN = '''
+    [
+        {
+            "name":             name of repository
+
+            "full_name":        full name of repository
```
Comment (klfgit) [inline plugins/modules/repository_information.py]: Specify the type as done for the other fields.

File: plugins/modules/repository_information.py [hunk 16]
```diff
@@ -0,0 +1,165 @@
...
+
+            "archived":         archived status of repository (bool: true or false),
+
+            "language":         language that the repo is using (as string),
+
+            "visibility":       for other users ("private" or "public"),
+
+            "url":              url for repo (as string),
+
+            "default_branch":   branch that repo defaults to (as string),
```
Comment (klfgit) [inline plugins/modules/repository_information.py]: "branch that repo defaults to" -> default branch of the repo

File: plugins/modules/repository_information.py [hunk 17]
```diff
@@ -0,0 +1,165 @@
...
+
+            "hooks_url":        url for hooks (as string),
+
+            "clone_url":        url for cloning (as string)
+        }
```
Comment (klfgit) [inline plugins/modules/repository_information.py]: } -> },

File: plugins/modules/repository_information.py [hunk 18]
```diff
@@ -0,0 +1,165 @@
...
+        {
+            ...
+        }
```
Comment (klfgit) [inline plugins/modules/repository_information.py]: } -> },

**Review by klfgit — CHANGES_REQUESTED** [review]
squash to a single commit.

File: .github/workflows/ansible-test.yml [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+        uses: ansible-community/ansible-test-gh-action@release/v1
+        with:
+          ansible-core-version: stable-2.11
+          pre-test-cmd: echo This runs before the ansible-test invocation
+          python-version: 3.9
+          target-python-version: 3.9
+          testing-type: units
+          test-deps: >-
+            ansible.netcommon
+            ansible.utils
```
Comment (klfgit) [inline .github/workflows/ansible-test.yml]: Run the unit test 1st.

File: .github/workflows/linter.yaml (hunk 1 above)
Comment (klfgit) [reply .github/workflows/linter.yaml]: If the lint checks fail the build should fail.

**Review by klfgit — CHANGES_REQUESTED** [review]

File: README.md [hunk 20]
```diff
@@ -1,2 +1,63 @@
...
+| Name                                                                                                                                             | Description                                                                            |
+| ------------------------------------------------------------------------------------------------------------------------------------------------ | -------------------------------------------------------------------------------------- |
+| [repository_info](https://github.com/senior-design-21-22/ansible-collection-github/blob/repo-information-module/docs/repository_information.rst) | Output repositories along with vital information from an (user specified) organization |
+
+## Usage
+
+### Repository Information (From organization)
+
+```
+  - name: "List GitHub repositories within org"
```
Comment (klfgit) [inline README.md]: List GitHub repositories within org -> "List GitHub repositories within an organization"

**Review by klfgit — CHANGES_REQUESTED** [review]

File: plugins/modules/repository_information.py (hunk 6 above)
Comment (jacobeicher) [reply plugins/modules/repository_information.py]: They are not available, I have not been able to find any reason as to why they unavailable

**Review by robert4man — COMMENTED** [review]
According to the Ansible documentation, docs files should be in Markdown:
https://docs.ansible.com/ansible/2.10/dev_guide/developing_collections.html#docs-directory

```
% ansible-doc ohioit.github.repository_information
ERROR! module ohioit.github.repository_information missing documentation (or could not parse documentation): while parsing a flow mapping
  in "<unicode string>", line 3, column 9:
            {
            ^
expected ',' or '}', but got ':'
  in "<unicode string>", line 6, column 24:
                "full_name":        full name of repository
                           ^
```

**Review by robert4man — COMMENTED** [review]
Has this code been tested with the GHE appliance? It worked for me against a cloud Org, but not an appliance Org. The documentation around enterprise_urlstring is confusing:

enterprise_url | Unecessary in event of user token | Enterprise url is necessary when module is recieving an enterprise token

The token and org are always required. I assumed enterprise_url would be used with the GHE appliance as opposed to a github.com org.

File: plugins/modules/repository_information.py [hunk 21]
```diff
@@ -0,0 +1,165 @@
...
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: repository_info
```
Comment (robert4man) [inline plugins/modules/repository_information.py]: shouldn't the module name match the file name?

**Pull request closed without merging**